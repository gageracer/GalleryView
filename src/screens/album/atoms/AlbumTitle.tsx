import React from 'react'
import { Text } from 'native-base'
import { StyleSheet, TextStyle } from 'react-native'
interface props {
    data: string
} 

export const AlbumTitle: React.FC<props> = ({ data }) => (
    <Text style={styles.text}>
        {data}
    </Text>
)

const styles = StyleSheet.create({
    text: {
        marginLeft:  25,
        fontSize: 18,
    } as TextStyle,
})