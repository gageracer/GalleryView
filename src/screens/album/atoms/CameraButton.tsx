import React from 'react'
import { View } from 'native-base'
import { StyleSheet } from 'react-native'
import { useHistory } from "react-router-dom"
import { MaterialCommunityIcons } from '@expo/vector-icons';

export const CameraButton: React.FC = () => {
    const history = useHistory()

    const goCamera = () => {
        history.push("/camera")
    }
    
    return (
        <View  style={styles.button}>
            <MaterialCommunityIcons onPress={goCamera} name="camera-plus" size={32} color="white" />
        </View>
    )
}

const styles = StyleSheet.create({
    button: {
        zIndex: 10,
        position: 'absolute',
        bottom: 45,
        right: 45,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#2c2c51',
        width: 64,
        height: 64,
        borderRadius: 50,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 1,
        },
        shadowOpacity: 0.22,
        shadowRadius: 2.22,

        elevation: 3,
    },
})