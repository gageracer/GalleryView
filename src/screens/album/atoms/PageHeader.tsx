import React from 'react'
import { Header, View, Button, Icon } from 'native-base'
import { StyleSheet } from 'react-native'

export const PageHeader: React.FC = () => { 
    return(
        <View>
            <Header androidStatusBarColor={"#303055"} style={styles.header}/>
            <View style={styles.right}>
                <Button transparent>
                    <Icon style={styles.icon} name='menu' />
                </Button>
            </View>
        </View>
        )
}


const styles = StyleSheet.create({
    header: {
        height: 50,
        backgroundColor: '#303055'
    },
    right: {
        flexDirection: 'row-reverse',
    },
    icon: {
        color: 'black'
    }
})