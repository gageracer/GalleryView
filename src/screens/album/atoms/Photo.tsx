import React from 'react'
import { StyleSheet, Image, ImageStyle, ImageSourcePropType } from 'react-native'

interface props {
    data: ImageSourcePropType
} 

export const Photo: React.FC<props> = ({data}) => {
    return (
        <Image
        style={styles.image}
            source={data} />
    )
}

const styles = StyleSheet.create({
    image: {
        borderRadius: 2,
        width: '100%',
        resizeMode: 'contain',
    } as ImageStyle,
    
})