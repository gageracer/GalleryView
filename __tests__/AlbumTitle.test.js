import React from 'react';
import renderer from 'react-test-renderer';

import {AlbumTitle} from '../src/screens/album/atoms/AlbumTitle';

describe('<AlbumTitle />', () => {
    it('has 1 child', () => {
        const tree = renderer.create(<AlbumTitle data={'test'} />).toJSON();
        expect(tree.children.length).toBe(1);
    })
    it('renders correctly', () => {
        const tree = renderer.create(<AlbumTitle data={'tesdt'}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

