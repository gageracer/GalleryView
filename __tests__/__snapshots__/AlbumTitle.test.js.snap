// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlbumTitle /> renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      },
      Object {
        "fontSize": 18,
        "marginLeft": 25,
      },
    ]
  }
  uppercase={false}
>
  tesdt
</Text>
`;
